dom/src/main/java/org/apache/james/mime4j/message/MessageBuilder.java

```diff
@@ -75,15 +75,68 @@
     private BodyFactory bodyFactory;
     private boolean flatMode;
     private boolean rawContent;
+    private final boolean ensureDateNotNull;
```

Comment:
1. Me neither
2. There's other ways than deprecation:
  - Post a mail on MIME4J user list
  - Write a blog post on the site
  - Issue a WARNING log when default date is used
  - ...

Sorry, but I value API stability a lot, when it comes to things people are using.